utils/analyticsmetrics.go

```diff
@@ -0,0 +1,204 @@
+	finalizeEvent *xscservices.XscAnalyticsGeneralEventFinalize
+}
+
+func NewAnalyticsMetricsService(serviceDetails *config.ServerDetails) *AnalyticsMetricsService {
+	ams := AnalyticsMetricsService{}
+	xscManager, err := CreateXscServiceManager(serviceDetails)
+	if err != nil {
+		// When an error occurs, shouldReportEvents will be false and no XscServiceManager commands will be executed.
+		log.Debug(fmt.Sprintf("Failed to create xsc manager for analytics metrics service. %s", err.Error()))
+		return &ams
+	}
+	ams.xscManager = xscManager
+	ams.shouldReportEvents = ams.calcShouldReportEvents()
+	return &ams
+}
+
+func (ams *AnalyticsMetricsService) calcShouldReportEvents() bool {
+	// A user who explicitly requests not to send reports will not receive XSC analytics metrics.
+	if !usage.ShouldReportUsage() {
+		return false
+	}
+	// Verify xsc version.
+	xscVersion, err := ams.xscManager.GetVersion()
+	if err != nil {
+		return false
+	}
+	if err = clientutils.ValidateMinimumVersion(clientutils.Xsc, xscVersion, xscservices.AnalyticsMetricsMinXscVersion); err != nil {
+		return false
+	}
+	return true
+}
+
+func (ams *AnalyticsMetricsService) XscManager() *xsc.XscServicesManager {
+	return ams.xscManager
+}
+
+func (ams *AnalyticsMetricsService) SetMsi(msi string) {
+	ams.msi = msi
+}
+
+func (ams *AnalyticsMetricsService) GetMsi() string {
+	return ams.msi
+}
+
+func (ams *AnalyticsMetricsService) SetStartTime() {
+	ams.startTime = time.Now()
+}
+
+func (ams *AnalyticsMetricsService) GetStartTime() time.Time {
+	return ams.startTime
+}
+
+func (ams *AnalyticsMetricsService) ShouldReportEvents() bool {
+	return ams.shouldReportEvents
+}
+
+func (ams *AnalyticsMetricsService) FinalizeEvent() *xscservices.XscAnalyticsGeneralEventFinalize {
+	return ams.finalizeEvent
+}
+
+func (ams *AnalyticsMetricsService) SetFinalizeEvent(finalizeEvent *xscservices.XscAnalyticsGeneralEventFinalize) {
+	ams.finalizeEvent = finalizeEvent
+}
+
+func (ams *AnalyticsMetricsService) CreateGeneralEvent(product xscservices.ProductName, eventType xscservices.EventType) *xscservices.XscAnalyticsGeneralEvent {
+	osAndArc, err := coreutils.GetOSAndArc()
+	curOs, curArch := "", ""
+	if err != nil {
+		log.Debug(fmt.Errorf("failed to get os and arcitucture for general event request to XSC service, error: %s ", err.Error()))
+	} else {
+		splitOsAndArch := strings.Split(osAndArc, "-")
+		curOs = splitOsAndArch[0]
+		curArch = splitOsAndArch[1]
+	}
+
+	event := xscservices.XscAnalyticsGeneralEvent{
+		XscAnalyticsBasicGeneralEvent: xscservices.XscAnalyticsBasicGeneralEvent{
+			EventType:              eventType,
+			EventStatus:            xscservices.Started,
+			Product:                product,
+			JfrogUser:              ams.xscManager.Config().GetServiceDetails().GetUser(),
+			OsPlatform:             curOs,
+			OsArchitecture:         curArch,
+			AnalyzerManagerVersion: GetAnalyzerManagerVersion(),
```

Comment:
GetAnalyzerManagerVersion will soon return `[RELEASE]` default value. I think it is neccessery to implement running the command `version` on the analyzerManager binary.

Reply:
The person who will implement it in the future will do so when it is relevant.